### DISCUSSION TITLE
Try to use the exported Converter class to customize the function.

### DISCUSSION THREAD

**Discussion by Saktawdi** [marp-team/marp#464] [Q&A] [answered]
like the title, I mainly use marp-core, but one headache is that I cannot export pptx.
I have tried to use some dependencies to make html generate pdf, but because marp's css is difficult to obtain, the pdf effect is not ideal.
Therefore, my eyes turned to the marp-cli. After reading the source code, I found that the Converter class  should be able to achieve the function I wanted: output pptx files through md strings or html after marp.rander.
This is why I want to come to the community for help, because I am not familiar with ts, and hope to **seek some help with the use of Converter classes** here.Let the function become util.

The worst case is to use exec to use the marp-cli command.

**Comment by yhatt** [comment]
In the first place, Marp CLI's [`Converter` class](https://github.com/marp-team/marp-cli/blob/main/src/converter.ts) is _for internal_, and it is not exported to `@marp-team/marp-cli` npm package.

Since the Converter class is closely related to tool's behavior, it is designed to be used just through the command-line interface. Therefore, CLI users cannot use `Converter' directly.

[The documentation about public API is here](https://github.com/marp-team/marp-cli#api-experimental). It can invoke Markdown conversion without using `exec`.

```javascript
import { marpCli } from '@marp-team/marp-cli'

// Convert a.md to a.pptx
await marpCli(['a.md', '-o', 'a.pptx'])
```

**Comment by yhatt** [comment, marked as answer]
FYI following are simplify version how to create PPTX from Marp Markdown. Hope to help for creating your own PPTX conversion logic :)

1. Generate the pair of HTML snippet and CSS by `@marp-team/marp-core`.
2. Create an standalone HTML file by combining them
3. Load created HTML using browser automation library (e.g. Puppeteer, Playwright)
4. Generate screenshots per page
5. Create PPTX based on them, powered by PPTX generation library (e.g. PptxGenJS, officegen)
